BuildProject/App.tsx

```diff
@@ -32,117 +30,131 @@ const App: () => Node = () => {
+    recordTestPassed('logAddToCartAction');
 
     PromotedMetrics.logRemoveFromCartAction(content);
-    logPassed('logRemoveFromCartAction');
+    recordTestPassed('logRemoveFromCartAction');
 
     PromotedMetrics.logCheckoutAction();
-    logPassed('logCheckoutAction');
+    recordTestPassed('logCheckoutAction');
 
     PromotedMetrics.logPurchaseAction(content);
-    logPassed('logPurchaseAction');
+    recordTestPassed('logPurchaseAction');
 
     PromotedMetrics.logShareAction(content);
-    logPassed('logShareAction');
+    recordTestPassed('logShareAction');
 
     PromotedMetrics.logLikeAction(content);
-    logPassed('logLikeAction');
+    recordTestPassed('logLikeAction');
 
     PromotedMetrics.logUnlikeAction(content);
-    logPassed('logUnlikeAction');
+    recordTestPassed('logUnlikeAction');
 
     PromotedMetrics.logCommentAction(content);
-    logPassed('logCommentAction');
+    recordTestPassed('logCommentAction');
 
     PromotedMetrics.logMakeOfferAction(content);
-    logPassed('logMakeOfferAction');
+    recordTestPassed('logMakeOfferAction');
 
     PromotedMetrics.logAskQuestionAction(content);
-    logPassed('logAskQuestionAction');
+    recordTestPassed('logAskQuestionAction');
 
     PromotedMetrics.logAnswerQuestionAction(content);
-    logPassed('logAnswerQuestionAction');
+    recordTestPassed('logAnswerQuestionAction');
 
     PromotedMetrics.logCompleteSignInAction();
-    logPassed('logCompleteSignInAction');
+    recordTestPassed('logCompleteSignInAction');
 
     PromotedMetrics.logCompleteSignUpAction();
-    logPassed('logCompleteSignUpAction');
+    recordTestPassed('logCompleteSignUpAction');
 
     PromotedMetrics.logAction('custom');
-    logPassed('logAction');
+    recordTestPassed('logAction');
 
     PromotedMetrics.logActionWithType('custom', ActionType.Navigate);
-    logPassed('logActionWithType');
+    recordTestPassed('logActionWithType');
 
     PromotedMetrics.logActionWithContent('custom', ActionType.Share, content);
-    logPassed('logActionWithContent');
+    recordTestPassed('logActionWithContent');
+  }
+
+  const testCollectionView = (recordTestPassed) => {
+    PromotedMetrics.collectionViewDidMount('hello');
+    recordTestPassed('collectionViewDidMount');
+
+    PromotedMetrics.collectionViewDidChange([], 'hello');
+    recordTestPassed('collectionViewDidChange');
+
+    PromotedMetrics.collectionViewWillUnmount('hello');
+    recordTestPassed('collectionViewWillUnmount');
+  }
 
+  const testAncestorIds = (recordTestPassed) => {
     PromotedMetrics.getCurrentOrPendingAncestorIds();
-    logPassed('getCurrentOrPendingAncestorIds');
+    recordTestPassed('getCurrentOrPendingAncestorIds');
 
     const ancestorIds = {
       logUserId: 'batman',
       sessionId: 'gotham',
       viewId: 'joker'
     } as AncestorIds;
     PromotedMetrics.setAncestorIds(ancestorIds);
-    logPassed('setAncestorIds');
+    recordTestPassed('setAncestorIds');
+  }
+
+  const handleTestAll = () => {
```

Comment:
nit - Might be easier to read if handleTestAll is above the other specific tests.

Reply:
Done.